algorithms/rainbow/agent.py

```diff
@@ -0,0 +1,158 @@
+"""
+Adapted from https://github.com/Kaixhin/Rainbow
+
+Wrapper class including setup, training and evaluation functions for Rainbow DQN model
+"""
+import os
+import numpy as np
+import torch
+from torch import optim
+
+from algorithms.rainbow.model import RainbowDQN
+
+
+class Agent:
+    """
+    Wraps control between both online and target network for setup, training and evaluation
+    """
+    def __init__(self, args, env):
+        """
+        Q(s,a) is the expect reward. Z is the full distribution from which Q is generated
+        Support represents the support of Z distribution
+        Z is represented with a fixed number of "atoms", which are discrete positions equidistant
```

Comment:
I don't think the atoms are the discrete positions themselves since the blog says "the positions of the atoms"
They're more like the discrete pairs of position and probability (x1,p1),…,(xN,pN)